src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs

```diff
@@ -72,6 +72,21 @@ public void ReadFileOver2GB()
             Assert.Throws<IOException>(() => File.ReadAllBytes(path));
         }
 
+        [Fact]
+        [OuterLoop]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/45954", TestPlatforms.Browser)]
+        public void ReadFileOverMaxArrayLength()
+        {
+            string path = GetTestFilePath();
+            using (FileStream fs = File.Create(path))
+            {
+                fs.SetLength(Array.MaxLength + 1L);
+            }
+
+            // File is too large for ReadAllBytes at once
+            Assert.Throws<IOException>(() => File.ReadAllBytes(path));
```

Comment:
Do we need a similar test for ReadAllBytesAsync?

Reply:
good point, I've added such test and also improved the test coverage for files that are non-seekable or report 0 length despite having non being empty